llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgExt/IR/LinalgExtOps.td

```diff
@@ -435,6 +435,55 @@ def IREELinalgExt_TopkOp : IREELinalgExt_Op<"topk",[
   }];
 }
 
+// TODO: is pack the right name? After all we are re-arranging the tensor.
+def IREELinalgExt_PackOp : IREELinalgExt_Op<"pack",[
```

Comment:
Do we consider adding pad semantics to the op and rename it to linalg_ext.pad_and_pack op?

Weigh in @MaheshRavishankar since he's been looking into it recently.

Reply:
Thanks @hanhanW for the feedbacks. I never thought about padding (Nicolas made me think about it). He suggested the following semantic: `tensor.pack == tensor.pad + tensor.expand_shape + tensor.transpose`. While for the expansion and the transpose it is clear what attributes we need for padding not so. Perhaps introducing a flag, similar to `nofold` in `PadOp` that tells the operation to pad along the blocked dimension if the dimension is not a multiple of the blocking factor? Or pass directly the padding to the operation like `PadOp` does?

Reply:
I think modeling on the pad op makes sense.
The nofold attribute will be useful, I would call it copy and blend it in the semantics, which will be in line with a more general way forward: https://discourse.llvm.org/t/rfc-structured-codegen-beyond-rectangular-arrays/64707#proposal-bufferization-copy-view-and-the-need-for-more-expressive-types-13

The pack name + optional padding region seem good enough for me to convey padding.
Renaming the op will be easy if it ever gets confusing.